cla-backend-go/gitlab_api/mr.go

```diff
@@ -139,22 +147,18 @@ func getUser(client *gitlab.Client, email, name *string) (*gitlab.User, error) {
 
 	if len(users) == 0 {
 		log.WithFields(f).Warnf("no user found for name : %s", *name)
-		return &gitlab.User{
-			Email: *email,
-			Name:  *name,
-		}, nil
+		return user, nil
 	}
 
 	// check if user exists for the given email
-	for _, user := range users {
-		log.WithFields(f).Debugf("checking user : %s for email : %s", user.Email, user.Email)
-		if user.Email == *email {
-			return user, nil
-		}
+	for _, found := range users {
```

Comment:
We have a for loop, but we grab the first and return it?  Why not have a check and return the first record?

```go
if len(users) > 0 {
  user.Username = found.Username
  user.ID = found.ID
  log.WithFields(f).Debugf("returning user: %+v", user)
  return user, nil
}
```